packages/@aws-cdk/aws-cognito/README.md

```diff
@@ -183,7 +183,12 @@ new UserPool(this, 'myuserpool', {
 As shown in the code snippet, there are data types that are available for custom attributes. The 'String' and 'Number'
 data types allow for further constraints on their length and values, respectively.
 
-Custom attributes cannot be marked as required.
+All custom attributes share the common properties `developerOnly` and `mutable`.
+
+ - `developerOnly` means that this attribute can only be modified by an administrator. The use of this property is discouraged
```

Comment:
It is Cognito's recommendation 

> We recommend that you use WriteAttributes in the user pool client to control how attributes can be mutated for new use cases instead of using DeveloperOnlyAttribute.

from [AWS::Cognito::UserPool SchemaAttribute in Cloudformation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cognito-userpool-schemaattribute.html)

Reply:
In such a case, should the CDK even implement this? Can we instead implement WriteAttributes on the client, so users use the right thing?

For users who really want to use this attribute or if they're migrating an existing user pool to the CDK, they can always access this via CDK's escape hatches.

Reply:
I like your point.

I think the implementation might involve changing the `UserPoolClient` class to allow set the read/write attributes for that specific client.

Do you think this should be implemented in its own issue/PR or is it fine to do it here?

Reply:
> Do you think this should be implemented in its own issue/PR or is it fine to do it here?

A separate PR would be preferable and cleaner.

Reply:
Ok, I will remove the support for the `developerOnly` attribute altogether and propose a new PR.